mkdocs.yml

```diff
@@ -12,21 +40,49 @@ markdown_extensions:
   # https://pypi.org/project/mdx-truly-sane-lists/
   - mdx_truly_sane_lists
 
+  - pymdownx.snippets
+
   # allows indented code blocks inside lists
-  - pymdownx.superfences
+  - pymdownx.superfences:
+      custom_fences:
+        - name: mermaid
+          class: mermaid
```

Comment:
#11001 moves us towards using D2 for diagrams; how does Mermaid fit into the picture?

Reply:
Mermaid is no longer in the photo :)

I determined mermaid diagrams were not powerful enough for the type of diagrams we wanted to show, and pivoted towards D2 instead.

Reply:
Gotcha, let's drop this from this PR in that case!